packages/lexical-yjs/src/SyncCursors.ts

```diff
@@ -423,41 +448,30 @@ export function syncCursorPositions(
         cursors.set(clientID, cursor);
       }
 
-      if (anchorPos !== null && focusPos !== null && focusing) {
-        const anchorAbsPos = createAbsolutePosition(anchorPos, binding);
-        const focusAbsPos = createAbsolutePosition(focusPos, binding);
-
-        if (anchorAbsPos !== null && focusAbsPos !== null) {
-          const [anchorCollabNode, anchorOffset] = getCollabNodeAndOffset(
-            anchorAbsPos.type,
-            anchorAbsPos.index,
-          );
-          const [focusCollabNode, focusOffset] = getCollabNodeAndOffset(
-            focusAbsPos.type,
-            focusAbsPos.index,
-          );
-
-          if (anchorCollabNode !== null && focusCollabNode !== null) {
-            const anchorKey = anchorCollabNode.getKey();
-            const focusKey = focusCollabNode.getKey();
-            selection = cursor.selection;
-
-            if (selection === null) {
-              selection = createCursorSelection(
-                cursor,
-                anchorKey,
-                anchorOffset,
-                focusKey,
-                focusOffset,
-              );
-            } else {
-              const anchor = selection.anchor;
-              const focus = selection.focus;
-              anchor.key = anchorKey;
-              anchor.offset = anchorOffset;
-              focus.key = focusKey;
-              focus.offset = focusOffset;
-            }
+      if (focusing) {
+        const {anchorCollabNode, anchorOffset, focusCollabNode, focusOffset} =
+          getAnchorAndFocusCollabNodesForUserState(binding, awareness);
+
+        if (anchorCollabNode !== null && focusCollabNode !== null) {
+          const anchorKey = anchorCollabNode.getKey();
+          const focusKey = focusCollabNode.getKey();
+          selection = cursor.selection;
+
+          if (selection === null) {
+            selection = createCursorSelection(
+              cursor,
+              anchorKey,
+              anchorOffset,
+              focusKey,
+              focusOffset,
+            );
+          } else {
+            const anchor = selection.anchor;
+            const focus = selection.focus;
+            anchor.key = anchorKey;
+            anchor.offset = anchorOffset;
+            focus.key = focusKey;
+            focus.offset = focusOffset;
```

Comment:
I understand that this is just copied from what was there before, but this should be using `$setPoint`  instead of setting the key and offset directly (setting of `type` is missing here). If the type of node changes this is incorrect and will cause weird bugs or crashes.

```suggestion
            $setPoint(selection.anchor, anchorKey, anchorOffset);
            $setPoint(selection.focus, focusKey, focusOffset);
```

Reply:
`selection.anchor` and `selection.focus` in the cursor selection aren't actual `Point`s and the `type` property isn't used anywhere where the cursor selection is used.

Reply:
Refactoring to use actual `Point`s is of course possible, but I think it'd be doing unnecessary work for something that's not used. For e.g., to initially create the point you'd need to get the anchor and focus nodes from the editor state to be able to check whether the type should be `element` or `type`.

Reply:
I tried refactoring to use Points [here](https://github.com/facebook/lexical/pull/6942/commits/d6f74b6923a2678faa59fbf633f4e6019a0f5f72), but have reverted that again since it seems to be making some extra changes like automatically deprecating and renaming syncCursorPositions because of rules-of-lexical lint rules.

Reply:
my fault, there's just not enough context the GitHub PR review UX for me to easily see that this kind of selection isn't a real selection, thanks for explaining!